app/routes/Account/containers/Account.js

```diff
@@ -10,29 +10,29 @@ class Account extends Component {
 
   handleShowQrCode = (data) => (e) => {
     createQrImage(
-      document.getElementById('qr'),
+      document.getElementById('qr-safe-address'),
       data,
       4
     )
   }
 
   render() {
-    const { address } = this.props.account
+    const { currentSafe } = this.props.safes
```

Comment:
I suggest instead of passing a big object to the container, create reselect selectors for memoize them and avoid extra re-renderings.

```getCurrentSafe```
```getSafes```

look good to me.

Reply:
Also, have you considered to store the safes as immutable objects? Arrays of objects are likely to mutate easily....